File: app/src/main/java/top/mikoto/sangnam/Receivers/BootReceiver.java
package top.mikoto.sangnam.Receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import java.util.ArrayList;

import top.mikoto.sangnam.Models.AlarmModel;
import top.mikoto.sangnam.Utils.AlarmManager;
import top.mikoto.sangnam.Utils.DB.DBHelper;

public class BootReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        //
        DBHelper dbHelper = new DBHelper(context,"ALARM",null,1);
        ArrayList<AlarmModel> alarmModels = dbHelper.getAllAlarms();
        AlarmManager alarmManager = new AlarmManager(context);
        for(AlarmModel alarmModel : alarmModels)
        {
            alarmManager.registerAlarm(alarmModel);
        }
        dbHelper.close();
    }
}

File: app/src/main/java/top/mikoto/sangnam/receivers/BootReceiver.kt
package top.mikoto.sangnam.receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import top.mikoto.sangnam.models.AlarmModel
import top.mikoto.sangnam.utils.AlarmManager
import top.mikoto.sangnam.utils.db.DBHelper
import java.util.*

class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        //
        val dbHelper = DBHelper(context, "ALARM", null, 1)
        val alarmModels: ArrayList<AlarmModel> = dbHelper.allAlarms
        val alarmManager = AlarmManager(context)
        for (alarmModel in alarmModels) {
            alarmManager.registerAlarm(alarmModel)
        }
        dbHelper.close()
    }
}